### ISSUE TITLE
During Cypress video recording in CI, will some times fail

### ISSUE THREAD

**Issue by lifeiscontent** [cypress-io/cypress#17455]
### Current behavior

When running `cypress run --parallel --record` sometimes video recording will fail and then both cypress dashboard and our CI will break.

CI will fail the step
cypress dashboard never completes a run and will never "complete"

```
Warning: We failed to record the video.
1163 |  
1164 | This error will not alter the exit code.
1165 |  
1166 | Error: ffmpeg exited with code 1: /********/src/packages/cypress2/cypress/videos/browser/auth/login.spec.ts.mp4: No such file or directory
1167 |  
1168 | at ChildProcess.<anonymous> (/root/.cache/Cypress/7.7.0/Cypress/resources/app/packages/server/node_modules/fluent-ffmpeg/lib/processor.js:182:22)
1169 | at ChildProcess.emit (events.js:315:20)
1170 | at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
1171
```

### Desired behavior

Cypress recording succeeds

### Test code to reproduce

not open source, sorry.

### Cypress Version

7.7.0

### Other

CI: Drone, running cypress via docker image

**Comment by lifeiscontent** [comment]
@jennifer-shehane can you take a peak at this? this is severely plaguing our cypress dashboard.

**Comment by jennifer-shehane** [comment]
@lifeiscontent 

- What browser are you recording in?
- How long are the specfiles that you are running when this happens? (Like normally they run in x seconds)?
- Could you run Cypress in [debug mode](https://on.cypress.io/troubleshooting#Print-DEBUG-logs) mode when this error throws and print the entire set of logs here?

**Comment by lifeiscontent** [comment]
@jennifer-shehane we're using the cypress/base image using electron.

from what I'm seeing in the dashboard this failure happens periodically. so I'm not sure how to run in debug mode because its not consistently failing.

**Comment by lifeiscontent** [comment]
@jennifer-shehane 
I put up a PR that uses `mkdirp` which looks to be the reason why this might be failing.

**Comment by lifeiscontent** [comment]
@jennifer-shehane bump

**Comment by davidungio** [comment]
I'm following this issue as well. Parallel testing will sometimes cause videos to fail to process.

**Comment by jennifer-shehane** [comment]
We believe this error is ocurring when running multiple Cypress instances in parallel. There's a larger explanation in this comment: https://github.com/cypress-io/cypress/issues/9128#issuecomment-894338562 

And we'll close this as a duplicate of #9128. 

Cypress wasn't originally built with the expectation of it being run on the same machine in parallel, so we'll have to come up with a larger solution for this. This would not be fixed by the work in #17469 unfortunately.

For now you should be able to set `trashAssetsBeforeRuns` to `false` to prevent this error from occuring or run each Cypress instance in a separate machine.
